File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.java
import java.io.*;

public class Main{
    public static void main(String[] args)throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine());
        int cnt = 0;
        int cnt6 = 0;
        int num = 666;
        int temp = 0;
        while (cnt < N){
            temp = num;
            cnt6 = 0;
            while(temp > 0){
                if(temp%10 == 6){
                    cnt6 +=1;
                }
                else{
                    cnt6 = 0;
                }

                if(cnt6 == 3){
                    cnt ++;
                    break;
                }
                temp = temp/10;
            }
            num += 1;
        }

        bw.write(String.valueOf(num - 1));
        bw.flush();
    }
}

File: 백준/Silver/1436. 영화감독 숌/영화감독 숌.cc
#include <stdio.h>

int main(){
    int N;
    scanf("%d",&N);
    int cnt = 0;
    int num = 666;
    int cnt_6=0;
    
    while(cnt < N){
        int temp = num;
        cnt_6=0;
        while(temp>0){
            if(temp%10==6){
                cnt_6+=1;
            }
            else{
                cnt_6=0;
            }
            if(cnt_6==3){
                cnt++;
                break;
            }
            temp = temp/10;
        }      
        num++;
    }
    
    printf("%d",num-1);
    
}